### DISCUSSION TITLE
Count number of rows on Entity

### DISCUSSION THREAD

**Discussion by ejaznoob** [SeaQL/sea-orm#1771] [Q&A]
I want to fetch number of records, I am using Postgres. Fetching the records and then calculating is not favorable for memory and performance reason.

Code 

             `  let total= TransactionEntity::find().count(&db).await.unwrap();`

Throws the error
```

  let total= TransactionEntity::find().count(&db).await.unwrap();
    |                                                     ^^^^^ `sea_orm::Select<transaction::Entity>` is not an iterator
```

**Comment by billy1624** [comment]
Hey @ejaznoob, it did not fetch all rows into the memory then do the counting. Instead, it count number of rows with a single `COUNT(*)` query. https://www.sea-ql.org/sea-orm-cookbook/021-count-select-result.html
